tests/components/bosch_shc/test_config_flow.py

```diff
@@ -0,0 +1,578 @@
+    entry = MockConfigEntry(
+        domain="bosch_shc", unique_id="test-mac", data={"host": "0.0.0.0"}
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.bosch_shc.config_flow.get_info_from_host",
+        return_value={
+            "title": "shc012345",
+            "unique_id": "test-mac",
+        },
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=DISCOVERY_INFO,
+            context={"source": config_entries.SOURCE_ZEROCONF},
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "already_configured"
+
+    # Test config entry got updated with latest IP
+    assert entry.data["host"] == "1.1.1.1"
+
+
+async def test_zeroconf_cannot_connect(hass):
+    """Test we get the form."""
+    with patch(
+        "boschshcpy.session.SHCSession.mdns_info", side_effect=SHCConnectionError
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=DISCOVERY_INFO,
+            context={"source": config_entries.SOURCE_ZEROCONF},
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "cannot_connect"
+
+
+async def test_zeroconf_not_bosch_shc(hass):
+    """Test we filter out non-bosch_shc devices."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        data={"host": "1.1.1.1", "name": "notboschshc"},
+        context={"source": config_entries.SOURCE_ZEROCONF},
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "not_bosch_shc"
+
+
+async def test_reauth(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="test-mac",
+        data={
+            "host": "1.1.1.1",
+            "hostname": "test-mac",
+            "ssl_certificate": "test-cert.pem",
+            "ssl_key": "test-key.pem",
+        },
+        title="shc012345",
+    )
+    mock_config.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_REAUTH},
+        data=mock_config.data,
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "reauth_confirm"
+
+    with patch(
+        "homeassistant.components.bosch_shc.config_flow.get_info_from_host",
+        return_value={
+            "title": "shc012345",
+            "unique_id": "test-mac",
+        },
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"host": "2.2.2.2"},
+        )
+
+        assert result2["type"] == "form"
+        assert result2["step_id"] == "credentials"
+        assert result2["errors"] == {}
+
+    with patch(
+        "homeassistant.components.bosch_shc.config_flow.create_credentials_and_validate",
```

Comment:
Please don't patch our code in the config flow. Patch in the periphery, eg the library, to avoid I/O.

Reply:
I already tested the code in another test, so I patched this to prevent duplicate code. I guess your remark is valid for all patches like "homeassistant.components.bosch_shc.config_flow..."?

Reply:
Yes, please always patch in the periphery. Sometimes it's ok to patch a different part of the integration but we don't want to patch the module we're aiming at testing.